enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,539 @@
+1. Infra Owner creates BareMetalHost resources with corresponding BMC credentials. They must be labeled so that they match the selctor on the InstallEnv and must be in the same namespace as the InstallEnv.
+1. A new controller, the Baremetal Agent Controller, sees the matching BareMetalHosts and boots them using the discovery ISO URL found in the InstallEnv's status.
+1. The Agent starts up on each host and reports back to the assisted service, which creates an Agent resource in the cluster. The Agent is automatically labeled with the labels that were specified in the InstallEnv's Spec.
+1. The Baremetal Agent Controller sets the Agent's Role field in its spec to "master" or "worker" if a corresponding label was present on its BareMetalHost.
+1. The Agent is automatically marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. Cluster Creator creates a ClusterDeployment describing a new cluster. It describes how many control-plane and worker agents to expect. It also includes a label selector to match Agent resources that should be part of the cluster.
+1. Cluster Creator applies a label to Agents if necessary so that they match the ClusterDeployment's selector.
+1. Once there are enough ready Agents of each role to fulfill the expected number as expressed on the ClusterDeployment, installation begins.
+
+#### Install Device
+
+The Agent resource Spec will include a field on which to specify the storage device
+where the OS should be installed. That field must be set by a platform-specific
+controller or some other actor that understands the underlying host.
+
+For bare metal, the BareMetalHost resource Spec already includes a
+RootDeviceHints section that will be utilized. The new Baremetal Agent
+Controller (previously described in the "Create Cluster" scenario) will use the
+BareMetalHost's RootDeviceHints and the Agent's discovery data to populate this
+field on the Agent.
+
+#### Day 2 Add Node Virtualmedia Multicluster
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke
+cluster. This assumes that a ClusterDeployment exists in the hub cluster.
+
+1. Infra Owner creates a BareMetalHost resource with a label that matches an InstallEnv selector.
+1. The Baremetal Agent Controller adds the discovery ISO URL to the BareMetalHost.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO on the BareMetalHost.
+1. The Agent starts up and reports back to the assisted service, which creates an Agent resource in the hub cluster. The Agent is labeled with the labels that were specified in the InstallEnv's Spec, optionally including a label to match the AgentSelector field on the ClusterDeployment.
```

Comment:
@avishayt it's not clear to me who is responsible to set the parameters on the host, in day1 we have bare-metal-agnet-controller. but i don't understand who is doing it in day 2 flow.